x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java

```diff
@@ -5121,6 +5141,62 @@ public void testLookupStats() {
         );
     }
 
+    public void testTranslateMetricsAggregate() {
+        assumeTrue("requires snapshot builds", Build.current().isSnapshot());
```

Comment:
maybe also unit test the case without grouping and when grouping by time_bucket + dimension?

Reply:
I added them in https://github.com/elastic/elasticsearch/pull/109979/commits/748b554a44f8fdebf0ea0591479368a73f8c191a.